SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/WorkerCoordinator.cs

```diff
@@ -0,0 +1,220 @@
+    ///     {port}                                  Receptionist port
+    ///     {worker_id}                             Worker id of the coordinator
+    ///     coordinator_start_delay_millis={value}  Minimum delay before the coordinator starts a simulated client
+    ///
+    /// All following arguments will be passed to the simulated player instance.
+    /// These arguments can contain the following placeholders, which will be replaced by the coordinator:
+    ///     `<IMPROBABLE_SIM_PLAYER_WORKER_ID>`     will be replaced with the worker id of the simulated player
+    ///     `<LOGIN_TOKEN>`                         will be replaced by the generated development auth login token
+    ///     `<PLAYER_IDENTITY_TOKEN>`               will be replaced by the generated development auth player identity token
+    ///
+    /// WORKER FLAGS
+    /// Additionally, the following worker flags are required for the coordinator:
+    ///     simulated_players_dev_auth_token_id     The development auth token id used to generate login tokens and player identity tokens
+    ///     simulated_players_target_deployment     Name of the target deployment which the simulated players will connect to
+    ///     target_num_simulated_players            The total number of simulated players that will connect to the target deployment. This value is used to determine the delay in connecting simulated players.
+    /// 
+    /// </summary>
+    internal static class WorkerCoordinator
+    {
+        // Arguments for the coordinator.
+        private const string START_DELAY_ARG = "coordinator_start_delay_millis";
+
+        // Argument placeholders for simulated players - these will be replaced by the coordinator by their actual values.
+        private const string WORKER_NAME_ARG = "<IMPROBABLE_SIM_PLAYER_WORKER_ID>";
+        private const string LOGIN_TOKEN_ARG = "<LOGIN_TOKEN>";
+        private const string PLAYER_IDENTITY_TOKEN_ARG = "<PLAYER_IDENTITY_TOKEN>";
+
+        // Worker flags.
+        private const string DEV_AUTH_TOKEN_ID_FLAG = "simulated_players_dev_auth_token_id";
+        private const string TARGET_DEPLOYMENT_FLAG = "simulated_players_target_deployment";
+        private const string NUM_SIM_PLAYERS_FLAG = "target_num_simulated_players";
+
+        private const string CoordinatorWorkerType = "SimulatedPlayerCoordinator";
+        private const string SimulatedPlayerWorkerType = "UnrealClient";
+        private const string SimulatedPlayerFilename = "StartSimulatedClient.sh";
+
+        private static Stack<Process> SimulatedPlayerList = new Stack<Process>();
+
+        private static Logger Logger = new Logger("/improbable/logs/WorkerCoordinator.log");
+        private const string LoggerName = "WorkerCoordinator.cs";
+        private const int ErrorExitStatus = 1;
+
+        private static Connection Connection;
+
+        private static Random Random = new Random();
+
```

Comment:
Just an idea for future work / ticket. There is no retry logic in any of the code so any failure is a single point of failure. Perhaps in the future we should look into doing retry logic with expotential backoffs with things like the connection and starting a client. :D

Reply:
Yeah fair point, would you like me to put in a ticket for better error handling?